src/v/storage/ntp_config.h

```diff
@@ -294,6 +298,47 @@ class ntp_config {
                           : cluster_default;
     }
 
+    std::optional<std::chrono::milliseconds> tombstone_retention_ms() const {
+        if (is_read_replica_mode_enabled()) {
+            // RRR sanity check.
+            return std::nullopt;
+        }
+        /*TODO(willem):
+          auto& cluster_default
+            = config::shard_local_cfg().tombstone_retention_ms();
+        */
+        if (_overrides) {
+            // Tombstone deletion should not be enabled at the same time as
+            // tiered storage.
+            if (
+              _overrides->shadow_indexing_mode.has_value()
+              && _overrides->shadow_indexing_mode.value()
+                   != model::shadow_indexing_mode::disabled) {
```

Comment:
curious why this is expressed as not-disabled rather than enabled?

Reply:
[model::shadow_indexing_mode::disabled](https://github.com/redpanda-data/redpanda/blob/97c11fea87c952ca9abc430eca26123d8d1c36e3/src/v/model/fundamental.h#L424) is a defined enumeration constant

Reply:
ahh makes sense